### PR TITLE
PanelEditor: Present actionable suggestions when panel cannot visualize current data

### PR DESCRIPTION
**Pull request by torkelo**
Reusing the action buttons from the "Add panel widget"  (refactored into new component) 

* Action to switch to table 
* Action to open visualization suggestions 

These actions are only visible in panel edit 

Screenshot from 2021-11-22 15-46-33

### CODE REVIEW

File: public/app/features/panel/components/PanelDataErrorView.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  const context = usePanelContext();
+  const builder = new VisualizationSuggestionsBuilder(props.data);
+  const { dataSummary } = builder;
+  const message = getMessageFor(props, dataSummary);
+  const dispatch = useDispatch();
+
+  const openVizPicker = () => {
+    store.setObject(LS_VISUALIZATION_SELECT_TAB_KEY, VisualizationSelectPaneTab.Suggestions);
+    dispatch(toggleVizPicker(true));
+  };
```
Comment (torkelo) [inline public/app/features/panel/components/PanelDataErrorView.tsx:26]: This is a bit hacky. But tried to make this part of panel edit redux state (like isVizPickerOpen) but required quite a number of changes and in the end it did not feel great either, Ideally, this should be done via URL param. But become too big a change for this PR

File: public/app/app.ts [hunk 2]
```diff
@@ -93,6 +95,7 @@ export class GrafanaApp {
       setLocale(config.bootData.user.locale);
       setWeekStart(config.bootData.user.weekStart);
       setPanelRenderer(PanelRenderer);
+      setPanelDataErrorView(PanelDataErrorView);
```
Comment (dprokop) [inline public/app/app.ts:98]: @torkelo why do we need to set the error view via runtime? Can't it be a component used the same way as you do down below in the TimeSeriesPanel? I can't see the getter function for the PanelDataErrorView used anywhere in this PR.

Comment (torkelo) [reply public/app/app.ts:98]: @dprokop not sure what you mean? How can it be used from plugins that way? 

There is no getter function, we export the PanelDataErrorView component from runtime (but implementation is in core). This is using the same way as we did for PanelRenderer

Comment (dprokop) [reply public/app/app.ts:98]: My bad, haven't noticed the PanelDataErrorView eport from runtime. I see now, we need it to live in core for it to be able to interact with redux etc. Forget the comment above :)

**Review by dprokop — APPROVED** [review]
The code looks good and I think this works great!

**Pull request merged**